src/Tokenizer/Tokens.php

```diff
@@ -1447,8 +1466,7 @@ private function getTokenNotOfKind(int $index, int $direction, callable $filter)
     {
         while (true) {
             $index += $direction;
-
-            if (!$this->offsetExists($index)) {
+            if (0 > $index || \count($this) <= $index) {
```

Comment:
how this will behave for non-list array ?

Reply:
Well, we have not met our interests in this PR, so I reverted these changes. I am still convinced we should require the data to be a list as this is the only tested variant and many methods are/were already broken for non-list input. But I do not want to object this nor spend more time on this one.